### PR TITLE
[CORE-1066] Refactor PachctlBashCommand

### PR DESCRIPTION
**Pull request by robert-uhl**
The existing `PachctlBashCommand` has implicit semantics: it only runs in a test; it can panic or fail the test; the config file is always stored under a name based on the test; state matters between runs.

This update adds new `PachctlBash` type which can be used to run scripts using `pachctl` in Bash. It can be run independently of a test; it always returns errors and never panics; the config file is stored in a caller-configurable path; state is preserved as long as the instance is not closed. A `Cmd` type which embeds `exec.Cmd` to provide easy access to buffered stdout/stdin is also provided.

### CODE REVIEW

File: src/internal/testutil/pachctl.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	b, err := io.ReadAll(cmd.stderr)
+	if err != nil {
+		panic(err)
+	}
+	return string(b)
+}
+
+func newCmd(ctx context.Context, name string, args []string, stdin io.Reader) Cmd {
+	var cmd Cmd
+	cmd.Cmd = exec.CommandContext(ctx, "/bin/bash", args...)
```
Comment (smalyala) [inline src/internal/testutil/pachctl.go:154]: Should "/bin/bash" be `name`?

File: src/internal/testutil/pachctl.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	if err != nil {
+		return errors.Wrap(err, "could not create new template")
+	}
+	if err := tmpl.Execute(w, data); err != nil {
+		return errors.Wrap(err, "could not execute script template")
+	}
+	return nil
+}
+
+// bashTemplate interprets scriptTemplate as an indented Go template for a Bash script; it returns an io.Reader containing
```
Comment (smalyala) [inline src/internal/testutil/pachctl.go:112]: Is this comment unfinished?

File: src/internal/testutil/cmds.go [hunk 3]
```diff
@@ -133,51 +136,37 @@ func BashCmd(cmd string, subs ...string) *exec.Cmd {
 	res.Env = os.Environ()
 	return res
 }
-func PachctlBashCmd(t *testing.T, c *client.APIClient, cmd string, subs ...string) *exec.Cmd {
-	t.Helper()
 
-	buf := new(bytes.Buffer)
-	_, err := buf.ReadFrom(bashCmd(cmd, subs...))
-	require.NoError(t, err)
+func subsToTemplateData(subs ...string) (map[string]string, error) {
```
Comment (smalyala) [inline src/internal/testutil/cmds.go:140]: Maybe we could have `bashCmd()` call this function instead of repeating the same steps.

**Review by smalyala — APPROVED** [review]
LGTM! Left a few comments

File: src/internal/testutil/pachctl.go (hunk 1 above)
Comment (robert-uhl) [reply src/internal/testutil/pachctl.go:154]: 🤦‍♂️

File: src/server/pfs/fuse/server_test.go [hunk 4]
```diff
@@ -745,7 +745,7 @@ func TestDeletingMountedRepo(t *testing.T) {
 		_, err := put("_mount", b)
 		require.NoError(t, err)
 
-		env.PachClient.DeleteProjectBranch(pfs.DefaultProjectName, "repo", "b1", false)
+		require.NoError(t, env.PachClient.DeleteProjectBranch(pfs.DefaultProjectName, "repo", "b1", false))
```
[the remark this answers is not in the document]
Comment (robert-uhl) [reply src/server/pfs/fuse/server_test.go:748]: It triggered `golangci-lint`, so I fixed it. Which raises the question of why the scripts which run `golangci-lint` didn’t error … maybe I should open another story for that, and then include this fix there.

**Review by armaanv — APPROVED** [review]

**Review by FahadBSyed — APPROVED** [review]

**Review by molinamelendezj — APPROVED** [review]

**Pull request merged**